ext/concurrent-ruby/com/concurrent_ruby/ext/JavaSemaphoreLibrary.java

```diff
@@ -54,7 +54,7 @@ public IRubyObject acquire(ThreadContext context, IRubyObject value) throws Inte
         @JRubyMethod
         public IRubyObject acquire(ThreadContext context, IRubyObject value, final Block block) throws InterruptedException {
             this.acquire(context, value);
-            block.yield(context, null);
+            block.yieldSpecific(context);
```

Comment:
I've replaced `yield(context, null)` with `yieldSpecific(context)` in the hopes that will fix the [error I was seeing in CI](https://github.com/ruby-concurrency/concurrent-ruby/runs/4826321806?check_suite_focus=true), however the [docs I was able to find on `Block`](https://www.javadoc.io/static/org.jruby/jruby-core/9.2.0.0/org/jruby/runtime/Block.html#yieldSpecific-org.jruby.runtime.ThreadContext-) provide no info as to the difference between them...

Looks like I need a maintainer to kick off another CI run though.

Reply:
> Looks like I need a maintainer...

Done - sorry not sure why that's set - it's just free GH Actions.

Reply:
Looks like it only limits first time contributors. Probably so I don't mine bitcoin or whatever...

Reply:
Looks like it's still failing, but I'm afraid I'm not familiar enough with JRuby to figure it out...

```

LoadError:
  Java extensions are required for JRuby.
  load error: concurrent/concurrent_ruby -- java.lang.RuntimeException: (LoadError) Mismatched block parameters for method com.concurrent_ruby.ext.JavaSemaphoreLibrary$JavaSemaphore.tryAcquire
# ./lib/concurrent-ruby/concurrent/utility/native_extension_loader.rb:35:in `load_native_extensions'
# ./lib/concurrent-ruby/concurrent/synchronization.rb:5:in `<main>'
# ./lib/concurrent-ruby/concurrent/atomic/event.rb:2:in `<main>'
# ./lib/concurrent-ruby/concurrent/concern/obligation.rb:4:in `<main>'
# ./lib/concurrent-ruby/concurrent/delay.rb:2:in `<main>'
# ./lib/concurrent-ruby/concurrent/configuration.rb:2:in `<main>'
# ./lib/concurrent-ruby/concurrent.rb:4:in `<main>'
# ./spec/spec_helper.rb:36:in `<main>'
# ------------------
# --- Caused by: ---
# Java::OrgJrubyExceptions::LoadError:
#   (LoadError) Mismatched block parameters for method com.concurrent_ruby.ext.JavaSemaphoreLibrary$JavaSemaphore.tryAcquire
#   RUBY.load_native_extensions(/home/runner/work/concurrent-ruby/concurrent-ruby/lib/concurrent-ruby/concurrent/utility/native_extension_loader.rb:35)
```

Unfortunately, even the detailed backtrace just points to the line loading the class.

Reply:
Okay, actually, I think I may have figured it out by looking at the [JRuby implementation of `IO.open`](https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/RubyIO.java#L1164-L1181), which does something similar to what we want to do here.

Looks like instead of overloading the methods (one with `final Block block`, and one without), I need to just have the one with the block argument and check `block.isGiven()`.

I'll push an update soon, but looking into that I realized it would probably be more idiomatic to return the result of the `yield`, so I'm going to change that too.